datafusion/physical-plan/src/sorts/streaming_merge.rs

```diff
@@ -143,8 +170,27 @@ impl<'a> StreamingMergeBuilder<'a> {
             fetch,
             expressions,
             enable_round_robin_tie_breaker,
+            spill_manager,
+            sorted_spill_files,
+            max_blocking_threads,
         } = self;
 
+        if spill_manager.is_some() && !sorted_spill_files.is_empty() {
+            return Ok(Box::pin(MultiLevelSortPreservingMergeStream::new(
+                spill_manager.unwrap(),
+                schema.clone().unwrap(),
+                sorted_spill_files,
+                streams,
+                expressions.clone(),
+                metrics.clone().unwrap(),
+                batch_size.unwrap(),
+                reservation.unwrap(),
+                max_blocking_threads,
+                fetch,
+                enable_round_robin_tie_breaker,
+            )?));
+        }
```

Comment:
Will remove all the unwrap. first want to know what do you think about the solution